software-engineering-policies/Containers/ContainerBestPracticies.md

```diff
@@ -0,0 +1,48 @@
+# Container Best Practices
+
+## Dockerfile Best Practices
+
+Docker provide a series of best practices for writing and developing Dockerfile’s. These are:
+
+* Best practices for [writing a Dockerfile](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/). 
+* Best practices for [developing with Docker](https://docs.docker.com/develop/dev-best-practices/).
+* Best practices for [developing with Docker](https://docs.docker.com/develop/security-best-practices/).
+
+The following items supersede these best practices:
+
+* Where a container needs to use secrets and passwords, these should be injected into the container via environment variables defined in the Dockerfile and should be encrypted at a minimum. 
+* Dockerfile’s should be linted to detect issues and vulnerabilities
+* Aim to use minimal dependencies, such as [distroless container base](https://github.com/GoogleContainerTools/distroless).
+* Avoid defining unnecessary privileges and avoid running as root.
+* Ensure there is a well-established process for updating dependencies.
+
+## Container Deployment Best Practices
+
+### Must Do
+
+* Where applicable at deployment, secrets must be stored via Azure KeyVault, AWS Secrets Manager, etc.
+* The containers resources, limits and requests for CPU and Memory consumption must be defined for deployment
+* Containers deployments and deployment environment must be configured using infrastructure as code.
+* Deployments must include a means to monitor the health of the container. For example, when using Kubernetes, configure readiness and liveliness probes.
+* Environments above Dev should have HTTPS enforced and Http to Http redirects enabled.
+* Logs generated by containers must be consumed.
+* Access to the underlying infrastructure running the container images must be blocked in all environments above Dev.
```

Comment:
You've lost the bullet point about container vulnerability scanning?

Reply:
This seemed to be sat in the list as a question. Happy to bring it back in. Do we run container scanning currently?

Reply:
Done!